scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1762,10 +1765,19 @@ class Router(formatOps: FormatOps) {
       case FormatToken(c: T.Comment, _, _) =>
         Seq(Split(getMod(formatToken), 0))
 
-      case FormatToken(_: T.KwImplicit, _, _)
+      case FormatToken(soft.ImplicitOrUsing(), _, _)
           if !style.binPack.unsafeDefnSite &&
-            !style.verticalMultiline.atDefnSite =>
-        opensImplicitParamList(prevNonComment(prev(formatToken))).fold {
+            !style.verticalMultiline.atDefnSite && isRightImplicitOrUsingSoftKw(
```

Comment:
can you insert a newline after `&&`? that way, you are likely to get a more compact, single-line formatting for
```
isRightImplicitOrUsingSoftKw(prev(formatToken), soft) =>
```